master/internal/experiment.go

```diff
@@ -168,6 +167,7 @@ func newExperiment(master *Master, expModel *model.Experiment, taskSpec *tasks.T
 	if agentUserGroup == nil {
 		agentUserGroup = &master.config.Security.DefaultTask
 	}
+	taskSpec.AgentUserGroup = agentUserGroup
```

Comment:
this snippet is in every task actor and would be really easily for someone to forget. may be nice to make it harder to miss.

Reply:
One of my next steps to move this defaulting logic to one place instead of separating out to everywhere. We will get there.

Reply:
sgtm